### ISSUE TITLE
plot_cpdb() error

### ISSUE THREAD

**Issue by rosanababu** [zktuong/ktplots#110]
Hi,

Many thanks for the amazing tool. I'm using ktplots and cellphoneDB for the first time. I've used cpdb_statistical_analysis_method from latest cellphoneDB v5 to run statistical analysis. plot_cpdb_heatmap() works fine on the results, unfortunately dot plot plot_cpdb() is showing error. Please could you help to solve the following error from plot_cpdb().

### Dot plot
> plot_cpdb(
+     scdata=sce, 
+     cell_type1="Pericyte",
+     cell_type2=".",  
+     celltype_key="Cell_type",
+     means=means_stat,
+     pvals=pvals_stat,
+     genes=c("TNFRSF13B", "STAT5A"),
+     title="interacting interactions!",
+ )
Error in `dplyr::left_join()`:
! Can't join `x$Var2` with `y$Var2` due to incompatible types.
ℹ `x$Var2` is a <integer>.
ℹ `y$Var2` is a <factor<4887b>>.
Run `rlang::last_trace()` to see where the error occurred.
> rlang::last_trace()
<error/dplyr_error_join_incompatible_type>
Error in `dplyr::left_join()`:
! Can't join `x$Var2` with `y$Var2` due to incompatible types.
ℹ `x$Var2` is a <integer>.
ℹ `y$Var2` is a <factor<4887b>>.
---
Backtrace:
    ▆
 1. └─ktplots::plot_cpdb(...)
 2.   ├─dplyr::left_join(df_means, df_pvals, by = c("Var1", "Var2"))
 3.   └─dplyr:::left_join.data.frame(df_means, df_pvals, by = c("Var1", "Var2"))
Run rlang::last_trace(drop = FALSE) to see 22 hidden frames.
> rlang::last_trace(drop = FALSE)
<error/dplyr_error_join_incompatible_type>
Error in `dplyr::left_join()`:
! Can't join `x$Var2` with `y$Var2` due to incompatible types.
ℹ `x$Var2` is a <integer>.
ℹ `y$Var2` is a <factor<4887b>>.
---
Backtrace:
     ▆
  1. ├─ktplots::plot_cpdb(...)
  2. │ ├─dplyr::left_join(df_means, df_pvals, by = c("Var1", "Var2"))
  3. │ └─dplyr:::left_join.data.frame(df_means, df_pvals, by = c("Var1", "Var2"))
  4. │   └─dplyr:::join_mutate(...)
  5. │     └─dplyr:::join_cast_common(x_key, y_key, vars, error_call = error_call)
  6. │       ├─rlang::try_fetch(...)
  7. │       │ └─base::withCallingHandlers(...)
  8. │       └─vctrs::vec_ptype2(x, y, x_arg = "", y_arg = "", call = error_call)
  9. ├─vctrs (local) `<fn>`()
 10. │ └─vctrs::vec_default_ptype2(...)
 11. │   ├─base::withRestarts(...)
 12. │   │ └─base (local) withOneRestart(expr, restarts[[1L]])
 13. │   │   └─base (local) doWithOneRestart(return(expr), restart)
 14. │   └─vctrs::stop_incompatible_type(...)
 15. │     └─vctrs:::stop_incompatible(...)
 16. │       └─vctrs:::stop_vctrs(...)
 17. │         └─rlang::abort(message, class = c(class, "vctrs_error"), ..., call = call)
 18. │           └─rlang:::signal_abort(cnd, .file)
 19. │             └─base::signalCondition(cnd)
 20. └─rlang (local) `<fn>`(`<vctrs__2>`)
 21.   └─handlers[[1L]](cnd)
 22.     └─dplyr:::rethrow_error_join_incompatible_type(cnd, vars, error_call)
 23.       └─dplyr:::stop_join(...)
 24.         └─dplyr:::stop_dplyr(...)
 25.           └─rlang::abort(...)

Kind regards,
RB

**Comment by zktuong** [comment]
hi @rosanababu , can i ask you to check if the column names for both your `means_stat` and `pvals_stat` are similar?

Also are the dimensions the same/similar?

**Comment by zktuong** [comment]
what if you remove the `genes` option?

**Comment by rosanababu** [comment]
> hi @rosanababu , can i ask you to check if the column names for both your `means_stat` and `pvals_stat` are similar?
> 
> Also are the dimensions the same/similar?

Hi, Thanks for your reply. Yes, the dimension and the column names for both means_stat and ovals_stat are similar. I could run it without any errors when I filtered out cells that has no UMIs using: seurat_subset <- subset(seurat, nCount_RNA > 10). Then, I created the metadata and counts matrices from seurat_subset to input CellPhoneDB. Now the outputs of CellphoneDB pvals_stat and means_stat is working fine even when I set splitkey for conditions. Thank you

**Comment by zktuong** [comment]
no more issues so i'm going to assume it working now.

**Comment by lucygarner** [comment]
I had the same error. I tracked the issue down to the following line of code in the `.prep_query_group` function:  

```
query <- row.names(data[data$interacting_pair %in% query_id, ])
```

The cause of the error was that I used tidyverse to read in the means table. As a result, the class was "tbl_df". Subsetting "data" (means dataframe) resulted in a loss of rownames, so query ended up as "1", "2", "3", etc. rather than the original rownames.

It would be helpful if a comment could be added to the vignettes to not use tidyverse to read in CellPhoneDB results.
